File: rust/examples/positions.rs
mod common;

use common::{get_keypair, login, market, rest_client, ExampleResult};

#[tokio::main]
async fn main() -> ExampleResult {
    let client = rest_client()?;
    let keypair = get_keypair()?;
    let session = login(&client, &keypair, false).await?;
    let wallet = session.user.trading_wallet(session.auth_method);
    let market = market(&client).await?;

    let all = client.positions().get(wallet).await?;
    let per_market = client
        .positions()
        .get_for_market(wallet, market.pubkey.as_str())
        .await?;

    println!("wallet: {}", wallet);
    println!("markets with positions: {}", all.total_markets);
    println!(
        "positions in {}: {}",
        market.slug,
        per_market.positions.len()
    );
    Ok(())
}

File: typescript/examples/positions.ts
import { tradingWallet } from "../src/auth";
import { restClient, getKeypair, login, market, runExample } from "./common";

async function main() {
  const client = restClient();
  const keypair = getKeypair();
  const session = await login(client, keypair);
  const wallet = tradingWallet(session.user, session.auth_method);
  const all = await client.positions().get(wallet);
  const m = await market(client);
  const perMarket = await client.positions().getForMarket(wallet, m.pubkey);
  console.log("wallet:", wallet);
  console.log("markets with positions:", all.total_markets);
  console.log(`positions in ${m.slug}: ${perMarket.positions.length}`);
}

void runExample(main);
